node/pkg/watchers/evm/connectors/poller.go

```diff
@@ -74,12 +74,15 @@ func (b *BlockPollConnector) run(ctx context.Context) error {
 }
 
 func (b *BlockPollConnector) pollBlocks(ctx context.Context, logger *zap.Logger, lastBlock *NewBlock) (lastPublishedBlock *NewBlock, retErr error) {
+	timeout, cancel := context.WithTimeout(ctx, 15*time.Second)
+	defer cancel()
```

Comment:
Are we concerned about any state that gets lost when we time out and the whole runnable is killed?  Would it be possible to rate limit on our end instead?  For example, we could buffer requests in a channel and have a separate goroutine send them out in a rate-limited manner.

Reply:
Agree, maybe we should be more graceful in L69 like the original version of the poller (pre my refactoring PR), allow X timeouts with some kind of backoff. This would automatically retry blocks that were missed.

Reply:
This change is being made to address an issue with Arbitrum in testnet because the provider we are using keeps disconnecting us. This should not happen in mainnet, and if it does, I feel it is reasonable for the runable to die. If we see the runable die in mainnet, we need to address the underlying cause, rather than preemptively code around the issue.

Reply:
Ack.  I'm fine with leaving this as is and we can revisit if it becomes a problem in the future but please add a comment here explaining the situation for future readers who don't have / forgot the context.

Reply:
Added a comment.